receiver/statsdreceiver/receiver.go

```diff
@@ -119,7 +119,8 @@ func (r *statsdReceiver) Start(ctx context.Context, host component.Host) error {
 					}
 				}
 			case metric := <-transferChan:
-				if err := r.parser.Aggregate(metric.Raw, metric.Addr); err != nil {
+				addr, _ := net.ResolveIPAddr(r.config.NetAddr.Transport, r.config.NetAddr.Endpoint)
```

Comment:
Please move this `ResolveIPAddr()` call into an initialization routine, so that it's not called on every packet.